mesonbuild/utils/posix.py

```diff
@@ -34,6 +34,9 @@ def __enter__(self) -> None:
         except (BlockingIOError, PermissionError):
             self.lockfile.close()
             raise MesonException('Some other Meson process is already using this build directory. Exiting.')
+        except OSError as e:
+            self.lockfile.close()
+            raise MesonException(e.strerror)
```

Comment:
The error message is just going to be "ERROR: Operation not supported"?

Maybe this should be something like:

```python
            raise MesonException('Failed to lock the build directory:', e.strerror)
```

Reply:
Yeah, that would be better.

Reply:
Updated.